CONTRIBUTING.md

```diff
@@ -2,73 +2,173 @@
 
 We welcome contributions to kURL. We appreciate your time and help. 
 
-# Development workflow
+## Supportability
 
-## Testing kURL
+kURL leverage on [kubeadin][kubeadmin]. Then, to test the project, you will need access to a `Linux/amd64`.
+You can check its [system requirements][kurl-system-requirements] for further info.
+
+## Pre-requirements for you local environment
+
+- npm (i.e. for mac os: `brew install npm`)
+- [Golang](https://go.dev/doc/install)
+- Docker
+
+## Development Workflow
+
+### Testing released versions
+
+*For packages that have already been released, you can save time by running `curl -L https://k8s.kurl.sh/dist/kubernetes-1.25.2.tar.gz | tar -xzv -C kurl -f -` and `curl -L https://k8s.kurl.sh/dist/docker-20.10.17.tar.gz | tar -xzv -C kurl -f -` on the test server.*  <br />
+
+- 📌 _Ensure that you have a_ **kurl** _directory already created (`mkdir kurl`) from wherever you run the aforementioned commands.
+*If using `containerd` building docker packages is not necessary. Instead to build packages run `make dist/containerd-1.6.8.tar.gz && tar xzvf dist/containerd-1.6.8.tar.gz` or to download already built packages `curl -L https://k8s.kurl.sh/dist/containerd-1.6.8.tar.gz | tar -xzv -C kurl -f -`*
+- 📌 *For centos/rhel hosts, `openssl` packages are required. Run `make dist/host-openssl.tar.gz && tar xzvf dist/host-openssl.tar.gz` or to download already built packages `curl -L https://k8s.kurl.sh/dist/host-openssl.tar.gz  | tar -xzv -C kurl -f -`*<br />
+- 📌 *In general, when testing local changes to an add-on, you'll need to install the host package(s) required for the particular add-on you're testing. E.g. if you want to install longhorn-1.2.4 with your local changes then you'll need to install the [required host packages](https://github.com/replicatedhq/kURL/blob/main/addons/longhorn/1.2.4/Manifest#L1-L4) prior to running the kURL installer.*
+
+### Testing kURL using Remote Server
+
+1. Set up a 'test server' to test kURL:
+
+Testing can be accomplished on systems capable of hosting supported container runtime. Local or remote Virtual Machine(s) or Instance(s) in a Public cloud provider.
+
+- [Using GCP](#using-gcp)
+- [Using Virtual Box on Mac](#virtual-box-on-mac-os)
+- [Using QEME on Mac](#QEMU-on-MacOS)
+
+**NOTE**: the remote host must have `rsync` binary installed.
+
+1. Ensure that you will build the images and code within `linux/amd64` platform.
+
+```
+   export GOOS=linux 
+   export GOARCH=amd64 # You do not need to export this one if your local env is amd64 already
+   export DOCKER_DEFAULT_PLATFORM=linux/amd64 # Required to ensure that the dependencies used in the Dockerfile images will be obtained from this platiform. Otherwise you might face issues like this [example](https://stackoverflow.com/questions/65612411/forcing-docker-to-use-linux-amd64-platform-by-default-on-macos)
+```
+
```

Comment:
```suggestion
**NOTE** The above configuration help you build and run the project from your local environment (mac M1/M2) and test it out on the remote server with `rsync`. However, in many scenarios you might find issues and you will need instead of that be connect via SSH into the codeserver and `rsync` with any instance that you might create in GCP.   
```